libwasmvm/src/querier.rs

```diff
@@ -81,12 +81,12 @@ impl Querier for GoQuerier {
         }
 
         let bin_result: Vec<u8> = output.unwrap_or_default();
-        let result = serde_json::from_slice(&bin_result).or_else(|e| {
-            Ok(SystemResult::Err(SystemError::InvalidResponse {
+        let sys_result: SystemResult<_> = serde_json::from_slice(&bin_result).unwrap_or_else(|e| {
+            SystemResult::Err(SystemError::InvalidResponse {
```

Comment:
Should behave the same 👍 
That type annotation isn't needed, but I'm fine with leaving it in for clarity.